apps/jobs/views.py

```diff
@@ -1361,6 +1361,79 @@ def delete_submission_message_from_queue(request, queue_name):
         return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
 
 
+@swagger_auto_schema(
+    methods=["get"],
+    manual_parameters=[
+        openapi.Parameter(
+            name="queue_name",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_STRING,
+            description="Queue Name",
+            required=True,
+        )
+    ],
+    operation_id="get_total_messages_in_queue",
+    responses={
+        status.HTTP_200_OK: openapi.Response(
+            description="dict object for count",
+            schema=openapi.Schema(
+                type=openapi.TYPE_OBJECT,
+                properties={
+                    "total_messages": openapi.Schema(
+                        type=openapi.TYPE_INTEGER,
+                        description="SQS message queue dict object",
+                    ),
+                },
+            ),
+        ),
+        status.HTTP_400_BAD_REQUEST: openapi.Response(
+            "{'error': 'Error message goes here'}"
+        ),
+    },
+)
+@api_view(["GET"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_total_messages_in_queue(request, queue_name):
+    """
+    API to fetch total number of messages in AWS SQS queue.
+
+    - Arguments:
+        ``queue_name``: AWS SQS queue name
+
+    - Returns:
+        ``total messages``: The total messages a key-value pair in a dict
+    """
+    try:
+        challenge = Challenge.objects.get(queue=queue_name)  # noqa
```

Comment:
Please `get_challenge_model` function here.

Reply:
`get_challenge_model` uses pk whereas this function takes in queue name as argument. If you want I can change this to pk and get queue name indirectly